### PR TITLE
feat(ui): remember pagination option, project search selection jumps into project

### PR DESCRIPTION
**Pull request by sergiofteixeira**
Hello, my first contribution.

This adds browser local storage to remember pagination option.

Also improves the search to show only what you searched for and when selection a project you jump into it.

### CODE REVIEW

File: ui/src/features/project/list/projects-list.tsx
Comment (Marvin9) [inline ui/src/features/project/list/projects-list.tsx:39]:
```suggestion
  const [pageSize, setPageSize] = useState(() => getStoredValue(PAGE_SIZE_KEY, 10));
```

File: ui/src/features/project/list/projects-list.tsx
Comment (Marvin9) [inline ui/src/features/project/list/projects-list.tsx:20]: Please re-use `useLocalStorage` hook in utils

File: ui/src/features/project/list/projects-list.tsx
Comment (Marvin9) [inline ui/src/features/project/list/projects-list.tsx:91]: You would need to add `showSizeChanger` option to show page size filter

File: ui/src/features/project/list/project-list-filter.tsx
Comment (Marvin9) [inline ui/src/features/project/list/project-list-filter.tsx:26]: I loved this accessibility addon

**Review by Marvin9 — CHANGES_REQUESTED** [review]
Couple of changes that needs to be in place before merging. Thanks for the PR!

File: ui/src/features/project/list/projects-list.tsx
Comment (sergiofteixeira) [reply ui/src/features/project/list/projects-list.tsx]: done

File: ui/src/features/project/list/projects-list.tsx
Comment (sergiofteixeira) [reply ui/src/features/project/list/projects-list.tsx]: please confirm if this is what you requested.

File: ui/src/features/project/list/project-list-filter.tsx
Comment (sergiofteixeira) [reply ui/src/features/project/list/project-list-filter.tsx:26]: thanks! 👍🏻

**Comment by sergiofteixeira** [comment]
Updated the pr with the changes requested

File: ui/src/features/project/list/projects-list.tsx
Comment (Marvin9) [reply ui/src/features/project/list/projects-list.tsx:63]: Yes it is what I expected. Thanks

**Comment by Marvin9** [comment]
There is just one bug indirectly blocking this PR that I am looking at ie. in local storage change the page size to the page which does not exist and refresh project list page

**Comment by sergiofteixeira** [comment]
> There is just one bug indirectly blocking this PR that I am looking at ie. in local storage change the page size to the page which does not exist and refresh project list page

  ```
useEffect(() => {
    if (data && page > Math.ceil(data.total / pageSize)) {
      setPage(Math.ceil(data.total / pageSize) || 1);
    }
  }, [data, page, pageSize, setPage]);
```
maybe use useEffect to check if data is available? else reset it. just an idea

**Comment by Marvin9** [comment]
@sergiofteixeira yes please push that change. I tested in local and it resolves the issue that I mentioned

**Comment by Marvin9** [comment]
Also don't forget to run `make lint-ui`

**Comment by sergiofteixeira** [comment]
@Marvin9 pushed those changes and ran lint

File: ui/src/features/project/list/projects-list.tsx
Comment (Marvin9) [inline ui/src/features/project/list/projects-list.tsx:32]: could you import this useEffect from React?

**Review by Marvin9 — APPROVED** [review]
One last change and then we can merge after CI checks. Thanks!

File: ui/src/features/project/list/projects-list.tsx
Comment (sergiofteixeira) [reply ui/src/features/project/list/projects-list.tsx]: my bad, imported it @Marvin9

**Comment by sergiofteixeira** [comment]
pushed the change @Marvin9

**Pull request merged**
